src/GameFace/Utils/UsersExtractions.cs

```diff
@@ -0,0 +1,91 @@
+﻿using GameFace.Controllers;
+using GameFace.Models;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace GameFace.Utils
+{
+    public static class UsersExtractions
+    {
+
+        public static int GetExperienceUserByTask(int user, int task)
+        {
+            using (var database = new GameFaceContext())
+            {
+                int experience = 0;
+                var users = database.Users.Include(x => x.xP).Where(x => x.id == user);
+                foreach (var u in users)
+                {
+                    experience += (from xp in u.xP where xp.idTask == task select database.Tasks.Single(t => t.id == xp.idTask) into tasks select tasks.value).Sum();
```

Comment:
I'm sure this can be simplified - I might be wrong by why are we checking the taskId of the XP as well as the taskId of the task. Surely if the task belongs to a
 Xp with taskId n, then the taskId of the task would be n.